tiling.py

```diff
@@ -25,25 +28,47 @@ def __init__(self, height, up, mirror1_ctr, mirror2_ctr):
             shapes.Plane(ctr1, ctr1 + up_proj, ctr1 + right_proj),
             shapes.Plane(ctr2, ctr2 + up_proj, ctr2 + right_proj)
         ]
+
+        b = ctr1 - up_proj * height * 0.5
+
+        self._ground_plane = shapes.Plane(b, b + m1n, b + right_proj)
+        self._ceiling_plane = shapes.Plane(b+ height * up_proj, b + right_proj+ height * up_proj, b + m1n+ height * up_proj)
+
         self._dihedral_transforms = [np.eye(4), utils.reflection_matrix(self._mirror_planes[0])]
+
+        #there are 4 fds in tfd with ceiling or floor mirror
+        if self._flag_ceiling or self._flag_floor:
+            reflect_plane = self._ground_plane if self._flag_floor else self._ceiling_plane
+            for tx in range(len(self._dihedral_transforms)):
+                self._dihedral_transforms.append(np.dot(self._dihedral_transforms[tx], utils.reflection_matrix(reflect_plane)))
+
+
         self._translational_basis = [2*m1n*width, 2*m2n*width]
-        b = ctr1 - up_proj * height * 0.5
+        if self._flag_ceiling or self._flag_floor:
+            self._translational_basis.extend([2 * height * up_proj, -2 * height * up_proj])
 
         self._vertices = [b - right_proj*width*0.5, b + right_proj*width*0.5,
                           b + right_proj*width*0.5 + m1n*width,
                           b - right_proj*width*0.5 + m1n*width]
 
         self._mirror_edges = [(0, 1), (2, 3)]
+
         tfdv = [b - right_proj*width*0.5, b + right_proj*width*0.5,
                 b + right_proj*width*0.5 + m1n*width*2.0,
                 b - right_proj * width * 0.5 + m1n*width*2.0]
+
+        #if only floor mirror, move tfd vertices downward
+        if self._flag_floor:
+            for i in range(len(tfdv)):
+                tfdv[i] = tfdv[i] - up_proj*height
+
```

Comment:
We don't need this anymore right? And if we do, then don't we end up with too many vertices?

Reply:
What I'm doing now is move all the vertices downward so that we can get correct vertices of tfd at the bottom face. If we don't do this, then we will have a tfd which will always above horizons.

Reply:
Without this codes, we will get something like the this:
![image](https://user-images.githubusercontent.com/6183742/29794746-2cc0491a-8bfe-11e7-8603-5bc5ac26ae85.png)
It's not correct because the tfd should be the fds on R2 and the ones reflected by the floor.